pytorch_forecasting/models/base/_base_model_v2.py

```diff
@@ -294,3 +294,175 @@ def log_metrics(
+        """
+        Standardize model outputs to a 4-dimensional tensor, with shape
+        (batch_size, timesteps, num_features, last_dim).
+
+        Parameters
+        ----------
+        prediction : torch.Tensor
+            The raw prediction tensor from the model.
+            - Must be a torch.Tensor (in the future, also accept a list of tensors for
+              multi-target forecasting).
+            - Supported dims: 2D, 3D or 4D tensors.
+            - if 2D: (batch_size, timesteps) - univariate forecasting
+            - if 3D:
+                a) (batch_size, timesteps, n_targets) - multivariate forecasting
+                b) (batch_size, timesteps, last_dim) - univariate forecasting with quantiles or distribution.
+                c) (batch_size, timesteps, n_targets * last_dim) - multivariate
+              forecasting with quantiles, where features and quantiles are flattened in dim 2.
+            - if 4D: (batch_size, timesteps, n_targets, last_dim) - multivariate
+              forecasting with quantiles or distribution parameters.
+            - In the future, once multi-target forecasting is supported, this
+              will also accept a list of tensors, where each tensor inside the list
+              is treated as above.
+            - If anything apart from the above dimensions is provided, an error is raised.
+
+        expected_dims : tuple[int, Optional[int], Optional[int], Optional[int]], default=(None, None, None, None)
+        A tuple specifying the dimensions: (n_targets, batch_size, timesteps, last_dim)
+
+            n_targets : int
+                - Position 0: Number of target features
+                - Must be provided explicitly (cannot be None)
+                - Used for reshaping 2D and 3D tensors to 4D.
+
+            batch_size : Optional[int], default=None
+                - Position 1: Expected batch size
+                - When specified: Validates prediction.shape[0]
+                - When None: Uses actual tensor dimension
+
+            timesteps : Optional[int], default=None
+                - Position 2: Expected number of timesteps
+                - When specified: Validates prediction.shape[1]
+                - When None: Uses actual tensor dimension
+
+            last_dim : Optional[int], default=None
+                - Position 3: Size of the last dimension.
+                - Common use case - quantile, sample, distribution params.
+                - When it is specified, it is used to directly reshape.
+                - When None and model uses QuantileLoss: It is set to the number of quantiles
+                - When None and no quantile information is available: It defaults to 1.
+                - If required, this can be extended to handle other cases where the last_dim is None
+                but its value can be inferred from the loss function or model configuration (apart from
+                the existing QuantileLoss case, of course).
+        Returns
```

Comment:
minor formatting remark:

* there should be a newline before headers
* there shuld be a newline before bullet point lists, and one after

(not consequential now, but this is an assumption when generating rst autodocs)